src/jrd/Monitoring.cpp

```diff
@@ -617,7 +623,55 @@ MonitoringSnapshot::MonitoringSnapshot(thread_db* tdbb, MemoryPool& pool)
 		}
 
 		if (store_record)
+		{
+			if (dbb->getEncodedOdsVersion() >= ODS_13_1)
+			{
+				FB_UINT64 stmtId;
+				StmtBlobs stmtBlobs;
+				dsc desc;
+
+				if ((rid == rel_mon_compiled_statements) && EVL_field(nullptr, record, f_mon_cmp_stmt_id, &desc))
+				{
+					stmtId = *(FB_UINT64*) desc.dsc_address;
+
+					if (EVL_field(nullptr, record, f_mon_cmp_stmt_sql_text, &desc))
+						stmtBlobs.text = *reinterpret_cast<bid*>(desc.dsc_address);
+					else
+						stmtBlobs.text.clear();
+
+					if (EVL_field(nullptr, record, f_mon_cmp_stmt_expl_plan, &desc))
+						stmtBlobs.plan = *reinterpret_cast<bid*>(desc.dsc_address);
+					else
+						stmtBlobs.plan.clear();
+
+					if (!stmtBlobs.text.isEmpty() || !stmtBlobs.plan.isEmpty())
+						blobsMap.put(stmtId, stmtBlobs);
+				}
+				else if ((rid == rel_mon_statements) && EVL_field(nullptr, record, f_mon_stmt_cmp_stmt_id, &desc))
```

Comment:
I'd add a comment that here we assume that compiled statements are stored in the dump before requests.